### PR TITLE
Fix a bug when "flutter build apk --build-number=2 --build-name=1.0.2" doesn't work as expected.

### PR DESCRIPTION
**Pull request by kangwang1988**
Currently, when using flutter build apk --build-number=2 --build-name=1.0.2 command to build apk with specific build number and names, the local.properties set properly could be reset because of 
call stack as below:
First set (works as expected, the build info is used and set)
```
 updateLocalProperties(project: project, buildInfo: buildInfo);
buildGradleProject
```

Second set(works unexpectedly for FlutterPluginVersion.managed and FlutterPluginVersion.v2)
```
updateLocalProperties
updateLocalProperties(project: flutterProject);
_gradleProject
_buildGradleProjectV2
buildGradleProject
```

When the second set is triggered, as the build info is missing, the build-number and name will be useless and the local.properties will be reset.

PS.
Though I want to write a test to avoid the later regression, I found it hard for me to do it in a concise way now.

### CODE REVIEW

**Comment by zoechi** [comment]
Is this supposed to fix #22788?

**Comment by kangwang1988** [comment]
@zoechi 
I guess so. The current behavior now seems to be a bug. The local.properties set priorly is reset later, which makes the user passed arguments not working.

**Comment by zoechi** [comment]
Is this overlapping with #23826 ?

File: packages/flutter_tools/lib/src/android/gradle.dart [hunk 1]
```diff
@@ -83,17 +83,17 @@ Future<File> getGradleAppOut(AndroidProject androidProject) async {
...
 }
 
 // Note: Dependencies are resolved and possibly downloaded as a side-effect
 // of calculating the app properties using Gradle. This may take minutes.
-Future<GradleProject> _readGradleProject() async {
+Future<GradleProject> _readGradleProject(BuildInfo buildInfo) async {
   final FlutterProject flutterProject = await FlutterProject.current();
   final String gradle = await _ensureGradle(flutterProject);
-  updateLocalProperties(project: flutterProject);
+  updateLocalProperties(project: flutterProject,buildInfo: buildInfo);
```
Comment (jason-simmons) [inline packages/flutter_tools/lib/src/android/gradle.dart]: style nit: add a space after the comma

File: packages/flutter_tools/lib/src/android/gradle.dart [hunk 2]
```diff
@@ -327,7 +327,7 @@ Future<void> _buildGradleProjectV2(
     String gradle,
     BuildInfo buildInfo,
     String target) async {
-  final GradleProject project = await _gradleProject();
+  final GradleProject project = await _gradleProject(buildInfo:buildInfo);
```
Comment (jason-simmons) [inline packages/flutter_tools/lib/src/android/gradle.dart]: style nit: add a space after the colon

**Review by jason-simmons — APPROVED** [review]

**Comment by the4thfloor** [comment]
To make it more stable, we should add `@required` for the `BuildInfo buildInfo` property of the `updateLocalProperties` method (line 214)

**Comment by kangwang1988** [comment]
@jason-simmons  Should I resolve the conflicts now?

**Comment by the4thfloor** [comment]
@kangwang1988 it looks that something went wrong with your last change(s).

As I remember there was a merge conflict. Have you resolved it with a rebase or with a merge from somewhere else?

The commit count jumped to 162, so I guess you merged master(?) into this commit?

To fix merge conflicts, you should do a [git rebse](https://docs.openstack.org/doc-contrib-guide/additional-git-workflow/rebase.html)

**Comment by kangwang1988** [comment]
@the4thfloor  I did a git rebase and resolve the conflicts.

**Comment by GregorySech** [comment]
Currently in stable `--build-number=x --build-name=a.b.c` doesn't override the version specified in pubspec.yaml is this pull request related to it or should I file a new issue?

**Comment by kangwang1988** [comment]
@GregorySech 
You can cherry-pick commit:`df9626c` to see if it solves your issue.

**Comment by xster** [comment]
Can you make sure your branch only has 04fb9a27ebea8147cd7937c3a1c3b3a3f22892a5 by rebasing on master instead of merging?

**Comment by kangwang1988** [comment]
This pr is not necessary anymore.
As in commit 693fb6560 by Stanislav, Dec 19, 2018, it's fixed. 
Closing it now.

**Pull request closed without merging**